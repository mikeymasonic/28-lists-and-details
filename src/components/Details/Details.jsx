import React from 'react';
import PropTypes from 'prop-types';

const Details = ({ name, image, occupation, status, description, portrayedBy }) => (
  <>
    <h3>{name}</h3>
    <img src={image} />
    <h4>{occupation}</h4>
    <h4>{status}</h4>
    <p>{description}</p>
    <p>{portrayedBy}</p>

  </>
);

Details.propTypes = {
  name: PropTypes.string,
  image: PropTypes.string,
  occupation: PropTypes.string,
  status: PropTypes.string,
  description: PropTypes.string,
  portrayedBy: PropTypes.string
};

export default Details;
