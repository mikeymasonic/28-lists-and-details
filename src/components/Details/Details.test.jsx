import React from 'react';
import { shallow } from 'enzyme';
import Details from './Details';

describe('Details component', () => {
  it('renders Details', () => {
    const wrapper = shallow(<Details 
      name="Dana Scully" 
      occupation="FBI Agent"
      status="Alive"
      image="https://vignette.wikia.nocookie.net/x-files/images/3/31/TXF_Scully.jpg/revision/latest/scale-to-width-down/310?cb=20171214040235"
      description="Dana Katherine Scully (Played by Gillian Anderson) is an FBI Special Agent and medical doctor with a background in hard science who brought her scientific expertise and perspective to the X-files."
      portrayedBy="Gillian Anderson"
    />);
    expect(wrapper).toMatchSnapshot();
  });
});
