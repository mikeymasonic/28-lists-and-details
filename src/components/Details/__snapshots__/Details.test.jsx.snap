// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details component renders Details 1`] = `
<Fragment>
  <h3>
    Dana Scully
  </h3>
  <img
    src="https://vignette.wikia.nocookie.net/x-files/images/3/31/TXF_Scully.jpg/revision/latest/scale-to-width-down/310?cb=20171214040235"
  />
  <h4>
    FBI Agent
  </h4>
  <h4>
    Alive
  </h4>
  <p>
    Dana Katherine Scully (Played by Gillian Anderson) is an FBI Special Agent and medical doctor with a background in hard science who brought her scientific expertise and perspective to the X-files.
  </p>
  <p>
    Gillian Anderson
  </p>
</Fragment>
`;
